### ISSUE TITLE
Add auto device discovery

### ISSUE THREAD

**Issue by santarl** [DvaMishkiLapa/VitaOxiPad#15]
Instead of adding ip address manually, have it auto detect running instances and offer to connect.

One approach would be to:

- host a mDNS server (or similar service) on the pc side (rust app) and have it broadcast itself
- have the vita detect it send a packet to it with its own ip
- which the rust app accepts and initiates the connection to.

I prefer the above approach instead of having the vita broadcast itself because idk if u can host a mDNS server on vita although that would makes things a lot simpler

**Comment by DvaMishkiLapa** [comment]
Yes, we can try a Broadcast packet on a network with special content like Heartbeat.

We can only hope that some ~~clever~~ Windows firewall doesn't interfere with this.

**Comment by DvaMishkiLapa** [comment]
On the other hand, it's not exactly safe.

Theoretically, someone could pretend to be a PS Vita device and the client would connect automatically.

There needs to be a mechanism to confirm and warn users so they don't mindlessly accept the first search option while sitting on a public network with 50ms latency xD

**Comment by santarl** [comment]
> Theoretically, someone could pretend to be a PS Vita device and the client would connect automatically.

we can use some kinda pin like moonlight / [sunshine ](https://github.com/LizardByte/Sunshine)does it
